### DISCUSSION TITLE
OpenCRVS 1.3 application deployment 3.3.6

### DISCUSSION THREAD

**Discussion by mihiruk** [opencrvs/opencrvs-core#6220] [Technical Q&A]
Hello Team,

Can you please guide me how to do that setting up the Environment secrets under the topic of 3.3.6,
bit confusing on adding the records to the known_hosts file, I'm able to ssh from my local env to node manager (ssh root@<my domain>), and my public key stored inside of the authorized_key in the path of /home/ubuntu/.ssh (/home/ubuntu/.ssh/ authorized_key) in the node manager, so no idea on why we need to create another file called known_hosts, can you briefly explain and guide me, please?

![open crvs](https://github.com/opencrvs/opencrvs-core/assets/34314471/06f7f4af-f9da-41cd-a301-207d89ff94e7)

**Comment by rikukissa** [comment]
Hi there! The purpose of this file is to allow Github Actions to verify the deployment target SSH host is really the same computer your own local machine reaches with the SSH address. So this variable is only used as part of automated deployment but doesn't have any significance once the environment is deployed.

You can see how the variable is used here:
https://github.com/opencrvs/opencrvs-farajaland/blob/develop/.github/workflows/deploy.yml#L63-L67

I personally tend to run this command locally and paste the output to the `KNOWN_HOSTS` Github Secret

```
ssh-keyscan -H my-opencrvs-instance.opencrvs.org,162.12.140.73
```

Let me know if this helps!

**Reply by mihiruk** [reply]
Hi Highly appreciated for your rapid response, i think I configured it, hope it will be working as expected,

I'm in the process of running the automated deployment into the targeted server, and i can't see the environment section in order to store the Environment secrets under 3.3.6, I googled and I found its a payable feature so, I added those Environment secrets inside of the global secret (image no1). and I ran the Deploy(production), and I got the error at the Deploy to: phase saying **Error: Argument --host is required** (image No 2), so then I tried to manually run via SSH, again I got the different issue mentioning "Error: Missing environment variable CONTENT_SECURITY_POLICY_WILDCARD." (image No 3)

can you kindly explain why those get occurred, and guide me to deploy it successfully pls?

thanks
![Image No 1](https://github.com/opencrvs/opencrvs-core/assets/34314471/f6687d7b-7ba5-43a9-80f6-b22d2f6e1257)
![Image No 2](https://github.com/opencrvs/opencrvs-core/assets/34314471/8e77608d-a8ab-4dd4-b165-306474a7d826)

**Reply by Zangetsu101** [reply]
Hi @mihiruk. Sorry to hear that you are having issues with the deployment. The main issue you are facing is likely two missing environment variables, notice how I said "environment variables" & not "environment secrets" which you can add from the variables tab in your first screenshot. We recently moved a couple of our environment secrets to environment variables, sorry that the documentation is not yet up to date about it.
The 2 variables are `DOMAIN` & `CONTENT_SECURITY_POLICY_WILDCARD`. You can learn about the `DOMAIN` variable here: https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment/3.3.6-deploy-automated-and-manual
Let's say the you have `DOMAIN=your-domain.com`, in most of the cases CONTENT_SECURITY_POLICY_WILDCARD will be `*.your-domain.com`. Hope this helps you!

**Reply by mihiruk** [reply]
Thank you so much for the instant reply, I will try and let you know.
Highly appreciated...

**Reply by mihiruk** [reply]
Hello team, 

I was able to deploy the OpenCRVS 1.3 on the target server (image no1), But i couldn't log into the basic services (ex : login.<your_domain>, openhim.<your_domain> etc ) eventhough those records are successfully resolved through the domain (image no 2), is it because [Letsencrypt](https://letsencrypt.org/) was not installed? or any possible reasons? 
note: i opened all the ports from the firewall level as a temp action. 
![Image 01](https://github.com/opencrvs/opencrvs-core/assets/34314471/8aab5604-f83e-462a-9033-9f957b31a4f7)
![Image 02](https://github.com/opencrvs/opencrvs-core/assets/34314471/a9d0725a-cc0c-42a5-9f09-7a12412b675e)

**Reply by mihiruk** [reply]
all the docker containers are running also on the target server
![Image 03](https://github.com/opencrvs/opencrvs-core/assets/34314471/ad8ee976-5248-40ef-a57c-fb38b02e7acb)

**Reply by mihiruk** [reply]
Hello Team,

Hope you have a good weekend!
Reminder on this and If you find any clue, please update us.
Im updating an SS of the services that deployed in the swarm herewith

Thanks
![Image 04](https://github.com/opencrvs/opencrvs-core/assets/34314471/3c489d81-396e-4d72-8ebe-43862913967c)

**Reply by mihiruk** [reply]
Hello Team,

Do u have any idea about this?

Thanks

**Reply by rikukissa** [reply]
Hi! I don't seem to be able to connect to http://login.slashsuite.com/ either. Is the server in an internal network you need a VPN to connect to? Let's encrypt only fetches the SSL certificates and you should be able to browse to login.slashsuite.com even before these are set up. 

![image](https://github.com/opencrvs/opencrvs-core/assets/1206987/4bb391b0-5c2b-4487-ab48-696ad55e5572)


Your Trafik doesn't seem to be running. Could you give me the output of 

```
docker service ps opencrvs_traefik --no-trunc
```

**Reply by mihiruk** [reply]
Hi, we deployed on AWS EC2 server which is attached to a firewall known as a security group. we opened a new rule and let reach pretty much all the traffic for the moment, I think there are no restrictions from the firewall level. I will attach an SS as well. also i could see few errors for above CMD, please find it

**Comment by mihiruk** [comment]
![Image 05](https://github.com/opencrvs/opencrvs-core/assets/34314471/feab5ff9-47a8-4720-9abe-5001faf13389)
![firewall](https://github.com/opencrvs/opencrvs-core/assets/34314471/2fdb2f79-d903-4681-869a-2f2e40234d6a)

**Reply by rikukissa** [reply]
Seems like there's already something else running in the port 80 🤔 What do you get if you run `lsof -i:80`

**Comment by mihiruk** [comment]
is it because of running nginx? 
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/9f7fff40-3c1d-48d0-9c2e-8f918ff40e38)

**Comment by mihiruk** [comment]
other day, nginx was installed mistakenly by us 🙄

**Reply by rikukissa** [reply]
Yea :) That must be it. Try removing it and then 

```
docker service scale opencrvs_traefik=0
docker service scale opencrvs_traefik=1
```

**Comment by mihiruk** [comment]
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/951a51fe-4bcf-451d-bc3a-430683359261)

**Reply by mihiruk** [reply]
let me try, could you please stay tuned with us until this will get solved pls
Thanks

**Reply by mihiruk** [reply]
Hi @rikukissa site is working for me when I changed my network, 
but no idea which username and password to use to log to the system, 🤔 can you help me out on this?

Also, I have another concern that our root domain (slashsuite.com) which we use for hosting the Opencrvs system also used for hosting several different services, hence we can't use login.<your_domain> type, in future. Are there any modifications that need to be done from the code level if domain records form as following 

login.opencrvs.slashsuite.com
kibana.opencrvs.slashsuite.com

**Comment by mihiruk** [comment]
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/0e4862ca-005b-4171-8e58-98a6d93be804)

failed service now up

**Reply by rikukissa** [reply]
Great! Next you most likely need to seed the database and then you should be able to log in.

**Comment by mihiruk** [comment]
but still im getting previous erros
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/ec4081e2-06c7-4ee5-a965-4bcc4eca4f63)

**Reply by rikukissa** [reply]
That's just the output showing what happened to the previously attempted containers. The top one is current which started successfully.

**Comment by mihiruk** [comment]
But isof ensured nginx was removed
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/cbcb45c3-d22c-4e49-9535-4abc81e60d00)

**Comment by mihiruk** [comment]
Thats great, still i cant access these sites, 😢

login.slashsuite.com
kibana.slashsuite.com
minio-console.slashsuite.com

**Reply by rikukissa** [reply]
This is what I see
![image](https://github.com/opencrvs/opencrvs-core/assets/1206987/70c30f3a-f13f-435b-b0a4-6bc119e557ca)
![image](https://github.com/opencrvs/opencrvs-core/assets/1206987/de6fbb8f-7950-40a8-a747-b102b02d650a)
![image](https://github.com/opencrvs/opencrvs-core/assets/1206987/1e8f3a97-b11d-4a7d-9998-04e8fad0c562)

**Comment by mihiruk** [comment]
Oh let me check pls

**Comment by mihiruk** [comment]
hi those are not working for me, but working well for my colleagues lol
could be the caching right?

**Reply by rikukissa** [reply]
Interesting.. could be 🤔 Can you try running `curl -L http://login.slashsuite.com/` locally? It should return an HTML page

**Comment by mihiruk** [comment]
yes its working, early it could only see once i logged into the container, (docker exec), nginx could be the reason.
its working locally for me, i will see whats cause of not working from the browser level

**Reply by rikukissa** [reply]
Sure! Glad you got it working

**Comment by mihiruk** [comment]
And thank you soo much, @Riku Rouvila you saved my time 
and other thing, we actually didn't add anything related to our own country, we just configured farajaland configuration as will be doing just a demo , later we will create our own country config, is it ok right?

**Reply by rikukissa** [reply]
That's perfectly fine 👍

**Comment by mihiruk** [comment]
Hi @rikukissa site is working for me when I changed my network,
but no idea which username and password to use to log to the system, 🤔 can you help me out on this?

Also, I have another concern that our root domain (slashsuite.com) which we use for hosting the Opencrvs system also used for hosting several different services, hence we can't use login.<your_domain> type, in future. Are there any modifications that need to be done from the code level if domain records form as following

login.opencrvs.slashsuite.com
kibana.opencrvs.slashsuite.com

**Reply by rikukissa** [reply]
I think there should not be any issue using subdomains like the ones you describe as long as the deployment is done with `HOSTNAME=opencrvs.slashsuite.com` and `CONTENT_SECURITY_POLICY_WILDCARD=*.opencrvs.slashsuite.com`.

The default users are created as part of a process called "seeding". You can read about it here: https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment/3.3.7-seeding-and-clearing-data-on-a-server

After you've done this, you should be able to login with the users defined in [this CSV file](https://github.com/opencrvs/opencrvs-farajaland/blob/develop/src/data-seeding/employees/source/default-employees.csv#L1).

**Comment by mihiruk** [comment]
Referring to the above concern about the username and password

![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/e0debd07-c8d9-477f-9f3f-24d24af1a1f2)

**Reply by euanmillar** [reply]
@mihiruk did you get any errors when seeding the database as @rikukissa described here:  https://github.com/opencrvs/opencrvs-core/discussions/6220#discussioncomment-7689725. Also "admin" is not one of the default testing users in the fake country "Farajaland" to log in as described here:  https://documentation.opencrvs.org/setup/3.-installation/3.2-set-up-your-own-country-configuration/3.2.4-set-up-employees-for-testing-or-production/3.2.3.1-prepare-source-file-for-test-employees. You are just using [default](https://github.com/opencrvs/opencrvs-core/discussions/6220#discussioncomment-7678928) users at this time correct?  @mihiruk we can offer you more dedicated support if you email us at team@opencrvs.org and please answer the following questions:  

Please tell us your full name and how you wish to be referred to.
Please tell us the name of the organisation or company you work for.
Are you working with a government on an active country implementation? ... If so, which country’s implementation are you working on?
Which of these describes your project best?

- Testing OpenCRVS for business development, educational purposes or for personal interest
- A research project to evaluate, configure and test OpenCRVS as a possible solution for a government, an active brief or proposal. (If so, tell us about your project, brief, the client and timeline if you can.)
- Actively installing and configuring OpenCRVS Core for production use in a country.
- Customising OpenCRVS Core code in production in a country

Want to build your skills and get your organisation formally accredited on the OpenCRVS platform? Learn more on the OpenCRVS Implementation Partner briefing calls on 30th November. 
There are 2 events planned for different time zones:
9am CET: https://lnkd.in/eeTURyKz 
4pm CET: https://lnkd.in/eTmnTDZv

**Reply by mihiruk** [reply]
@euanmillar Hello, yes @rikukissa helped a lot, i was able to resolve almost all the issues, now the system is working well as expected. 
I will write you for the questions raised.

**Reply by mihiruk** [reply]
@euanmillar I am Mihiruk and currently working as a DevOps engineer at Bevolv, which is a subsidiary of Inova IT Systems (Pvt) Ltd.As a preparation for the implementation partner program/session we have successfully deployed OpenCRVS 1.3. Our team will connect tomorrow for the session. Thank you so much!

**Reply by mihiruk** [reply]
Hello Team,

I was unable to re-seed data with customized default-employees.csv data. The seed action gets canceled by throwing an error message as mentioned bellow. 
Any idea?
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/8f06335b-d1f3-4df8-acca-71b8fde1218f)
![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/f18836ed-8b97-4b10-a478-1012de727894)

**Comment by euanmillar** [comment]
@mihiruk  You asked a question by email about OTP, adding other email addresses etc and this is explained in the [documentation](https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment/3.3.3-provision-a-comms-gateway).  

A comms gateway such as an SMTP Email API service is required to send OTP codes to real email addresses and this requires extensive configuration described in that step.  But as the documentation states, the code is six zeros **000000** during development.  

Regarding seeding, did you set the SUPER_USER_PASSWORD environment secret used when seeding as also described in the documentation.  https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment/3.3.6-deploy-automated-and-manual

May I suggest that you thoroughly refer to the docs again as I think all the answers you need are there.  In the Implementation Partner Programme we will train you step by step through this, so perhaps you should wait until then.   We will also share videos on YouTube for every step in case you are unable to join it.

**Reply by mihiruk** [reply]
Thanks for the feedback given, yeah we have configured the required configurations such as SMTP and such, the document really helped and everything is explained so neatly, but there are a few confusions, hope we can sort it down going forward. Now we are in the process of re-seeding customized user details, I think the SUPER_USER_PASSWORD issue got resolved. having another issue while the date is being seeded, I think it's a data formatting issue as I believe. hope it's not that much critical 

![image](https://github.com/opencrvs/opencrvs-core/assets/34314471/cf7c443a-99c7-43eb-92b7-ecc38eb00f25)
